API/application/iotjs.py

```diff
@@ -163,51 +158,52 @@ def build(self):
             # Enable memstat for IoT.js (libtuv, jerryscript, iotjs)
             self.apply_patches()
 
-        elif self.device.get_type() == 'rpi2' and self.os == 'linux':
-            build_flags.append('--target-board=%s' % self.device.get_type())
+        elif device.get_type() == 'rpi2':
+            build_flags.append('--target-board=%s' % device.get_type())
             build_flags.append('--jerry-cmake-param=-DFEATURE_VALGRIND_FREYA=ON')
             build_flags.append('--compile-flag=-g')
             build_flags.append('--jerry-compile-flag=-g')
 
-        elif self.device.get_type() == 'artik053' and self.os == 'tizenrt':
+        elif device.get_type() == 'artik053':
             build_flags.append('--target-board=artik05x')
             build_flags.append('--sysroot=%s' % paths.TIZENRT_OS_PATH)
 
         else:
             console.fail('Non-minimal IoT.js build failed, unsupported '
-                         'device-os (%s-%s) combination!' %
-                         (self.device.get_type(), self.os))
+                         'device (%s)!' % device.get_type())
 
         # Run the buildscript.
         utils.execute(paths.IOTJS_PATH, 'tools/build.py', build_flags)
 
+        os.build(self, self.buildtype, 'all')
```

Comment:
Maybe this can be moved to the `device.flash` method to be the app and the os builds independent.

Reply:
It would not be a right approach to call the `os.build` function in the `device.flash`. It must be independent from that. It seems that there are still some unresolved dependencies. We should talk about it offline and implements a better solution.